metals/src/main/scala/scala/meta/internal/rename/RenameProvider.scala

```diff
@@ -35,138 +33,152 @@ import scala.meta.internal.semanticdb.Synthetic
 import scala.meta.internal.semanticdb.SelectTree
 import scala.meta.internal.metals.MetalsServerConfig
 import scala.meta.internal.metals.ClientExperimentalCapabilities
+import scala.meta.pc.CancelToken
+import scala.concurrent.Future
+import scala.concurrent.ExecutionContext
 
 final class RenameProvider(
     referenceProvider: ReferenceProvider,
     implementationProvider: ImplementationProvider,
     definitionProvider: DefinitionProvider,
-    index: GlobalSymbolIndex,
     workspace: AbsolutePath,
     client: MetalsLanguageClient,
     buffers: Buffers,
     compilations: Compilations,
     metalsConfig: MetalsServerConfig,
     clientExperimentalCapabilities: ClientExperimentalCapabilities
-) {
+)(implicit executionContext: ExecutionContext) {
 
   private var awaitingSave = new ConcurrentLinkedQueue[() => Unit]
 
-  def prepareRename(params: TextDocumentPositionParams): Option[LSPRange] = {
+  def prepareRename(
+      params: TextDocumentPositionParams,
+      token: CancelToken
+  ): Future[Option[LSPRange]] = {
     if (!compilations.currentlyCompiling.isEmpty) {
       client.showMessage(isCompiling)
-      None
+      Future.successful(None)
     } else {
       val source = params.getTextDocument.getUri.toAbsolutePath
-      val symbolOccurence =
-        definitionProvider.symbolOccurrence(source, params.getPosition)
-      for {
-        (occurence, semanticDb) <- symbolOccurence
-        if canRenameSymbol(occurence.symbol, None)
-        range <- occurence.range
-      } yield range.toLSP
+      definitionProvider.definition(source, params, token).map { definition =>
+        val symbolOccurence =
+          definitionProvider.symbolOccurrence(source, params.getPosition)
+        for {
+          (occurence, _) <- symbolOccurence
```

Comment:
rename `symbolOccurence` -> `symbolOccurrence` let's fix this once and for all in this file :muscle:

Reply:
Fixed